### ISSUE TITLE
First item is focused on mobile

### ISSUE THREAD

**Issue by anezthes** [vaadin/vaadin-context-menu#266]
We have a `ContextMenu` that contains anchors. To make those anchors focusable with keyboard we had to add `tabindex="0"`. The downside is that the first anchor is focused on mobile as well.

Any fix and/or workaround available?

**Comment by jouni** [comment]
What’s the downside of having it focused?

**Comment by anezthes** [comment]
I'd say it's confusing and looks awful. When was the last time you opened up a menu on a mobile phone where any item had a focus ring on it? :)

**Comment by jouni** [comment]
Okay, that clears it up, that it’s the visible focus ring that you don’t want. I can agree with that for sure :)

**Comment by jouni** [comment]
Here’s a workaround (which I haven’t actually tried if it works):

```javascript
import { registerStyles, css } from '@vaadin/vaadin-themable-mixin/register-styles.js';

registerStyles('vaadin-item', css`
  @media (pointer: coarse) {
    :host([focus-ring]) {
      box-shadow: none;
    }
  }
`);
```

**Comment by jouni** [comment]
I think this is a broader issue in themes, that the focus-ring/focus-visible style should not be shown on touch devices, with some exceptions.

**Comment by knoobie** [comment]
Focus should always be visible, accessibility doesn't change between devices. 

My question would be; why is the first item focused? Why isn't the whole context-menu focused like it can be done with dialog / modal dialog? That could even create a nice focus-ring to the whole context-menu.

**Comment by jouni** [comment]
For touch devices, there usually isn’t a focus indicator in popup menus. For example:

![](https://www.appcoda.com/wp-content/uploads/2014/05/t11_12_ipad_colors_actionsheet.jpg)

Can’t remember if screen readers have some indicator, but I think that might come forced by the OS.

> why is the first item focused

That’s a good question. Focusing the overlay itself would be better. But again, in that case, I would not have a visible focus indicator, unless the context menu was opened with the keyboard.

**Comment by flefebure** [comment]
Hi,
Not on a touch device, and with flow, but is this "focus-ring" responsible of the weird blue border that appears on the first item of a menu (it looks only useful for a selection with the keyboard, not with the mouse)?

![image](https://user-images.githubusercontent.com/16290736/81127471-5401b900-8f0c-11ea-85c4-78d0ab6646fd.png)

If so, can you give me a flow workaround to hide it ?
Thanks

**Comment by jouni** [comment]
Yes, that’s the keyboard focus indicator. You should not remove that, as that makes it impossible to navigate the menu with the keyboard.

Not focusing the first item when the menu opens, and/or “[focus follows mouse](https://github.com/vaadin/vaadin-list-box/issues/14)” would solve this.

**Comment by flefebure** [comment]
Ok, eventually, can you give a little hint to how to style this ring, in a flow context ?

**Comment by jouni** [comment]
vaadin-item-styles.css
```css
:host([focus-ring]) {
  box-shadow: none !important;
}
```

```java
@CssImport(value = "./styles/vaadin-item-styles.css", themeFor = "vaadin-item")
```

That should get rid of the blue outline across all items, including the ones in Select, ListBox (and maybe ComboBox and MenuBar as well).

If you want to scope the style only to context menu, then you can use a custom property:

vaadin-item-styles.css
```css
:host([focus-ring]) {
  box-shadow: var(--item-focus-outline) !important;
}
```

vaadin-context-menu-overlay-styles.css
```css
:host {
  --item-focus-outline: none;
}
```

```java
@CssImport(value = "./styles/vaadin-item-styles.css", themeFor = "vaadin-item")
@CssImport(value = "./styles/vaadin-context-menu-overlay-styles.css", themeFor = "vaadin-context-menu-overlay")
```

**Comment by flefebure** [comment]
Thanks for the hour(s) saving tip

**Comment by mvysny** [comment]
To my eyes the focus indicator is confusing:
* I have no idea of its purpose since it only shows for the ContextMenu but not for other components.
* At first I thought that the first item was somehow selected or checked; however in combination with checked items it looks even weirder.
